### PR TITLE
gh-93166: Clarify http.server documentation using RFC 7230 terminology

### LINKED ISSUE #93166: Clarify http.server documentation using RFC 7230 terminology
The linked PR will clarify the descriptions of methods and attributes in the [`http.server` documentation](https://docs.python.org/3/library/http.server.html) using [RFC 7230](https://datatracker.ietf.org/doc/html/rfc7230) terminology.

It will also document the optional `directory` parameter of `CGIHTTPRequestHandler(request, client_address, server, directory=None)` inherited from `SimpleHTTPRequestHandler(request, client_address, server, directory=None)`.

### Linked PRs
* gh-30686

### PR DESCRIPTION
**Pull request by geryogam**
* Issue: gh-93166

### CODE REVIEW

**Comment by MaxwellDupre** [comment]
A nice piece of work to tidy this up. Only minor changes please:
Line 210:
`The *code* specifies the status code, *message* the reason phrase,`

Line 274:
`The *code* specifies the status code`

Adding `The` at the start of these lines. Otherwise it looks strange, since the italicisation is not that pronounced and looks like there is something missing.

**Comment by erlend-aasland** [comment]
This is way too large a change to pass as _skip issue_. Please file an issue for this change.

File: Doc/library/http.server.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ This module defines classes for implementing HTTP servers.
 
 One class, :class:`HTTPServer`, is a :class:`socketserver.TCPServer` subclass.
 It creates and listens at the HTTP socket, dispatching the requests to a
-handler.  Code to create and run the server looks like this::
+handler. Code to create and run the server looks like this::
```
Comment (merwok) [inline Doc/library/http.server.rst:29]: Please avoid this kind of changes.  They do not fix or improve anything, so just churn history for not benefit.

File: Doc/library/http.server.rst [hunk 2]
```diff
@@ -55,131 +55,131 @@ provides three different variants:
...
    specific to the request type. The method name is constructed from the
-   request. For example, for the request method ``SPAM``, the :meth:`do_SPAM`
+   request. For example, for the request method SPAM, the :meth:`do_SPAM`
    method will be called with no arguments. All of the relevant information is
-   stored in instance variables of the handler.  Subclasses should not need to
+   stored in instance variables of the handler. Subclasses should not need to
    override or extend the :meth:`__init__` method.
 
-   :class:`BaseHTTPRequestHandler` has the following instance variables:
+   :class:`BaseHTTPRequestHandler` defines the following instance variables:
```
Comment (merwok) [inline Doc/library/http.server.rst:73]: Are you disturbed by the wording that a class «has» instance variables?
I’m not sure that people would not understand what is meant here.  But it’s also a straightforward change, so why not.

File: Doc/library/http.server.rst [hunk 3]
```diff
@@ -55,131 +55,131 @@ provides three different variants:
...
 
    .. attribute:: client_address
 
-      Contains a tuple of the form ``(host, port)`` referring to the client's
+      A tuple of the form ``(host, port)`` referring to the client's
```
Comment (merwok) [inline Doc/library/http.server.rst:77]: I see how `contains` or `holds` may feel strange here, given that Python has names pointing to objects, not variables containing memory.  That could be the point of the ticket you’ve been asked to create, and a true (if small) improvement.

File: Doc/library/http.server.rst [hunk 4]
```diff
@@ -55,131 +55,131 @@ provides three different variants:
...
 
    .. attribute:: command
 
-      Contains the command (request type). For example, ``'GET'``.
+      The request method. For example, ``'GET'``.
 
    .. attribute:: path
 
-      Contains the request path. If query component of the URL is present,
+      The request target. If query component of the URL is present,
```
Comment (merwok) [inline Doc/library/http.server.rst:103]: Let’s not replace `path` with `target` here.

File: Doc/library/http.server.rst [hunk 5]
```diff
@@ -55,131 +55,131 @@ provides three different variants:
...
       then ``path`` includes the query. Using the terminology of :rfc:`3986`,
       ``path`` here includes ``hier-part`` and the ``query``.
 
    .. attribute:: request_version
 
-      Contains the version string from the request. For example, ``'HTTP/1.0'``.
+      The request HTTP version. For example, ``'HTTP/1.0'``.
```
Comment (merwok) [inline Doc/library/http.server.rst:109]: Previous wording was more elegant (except `contains`)

**Review by merwok — COMMENTED** [review]
I made a few notes on the first changes in the PR.

CPython is big and old, and as we’ve discussed in other PRs, sweeping changes are weighed cautiously.  There is also no goal to apply a very specific style to all parts of the documentation; it’s fine if different documents contributed by different authors may have different styles, as long as it fits in its overall doc section.

To apply your time and energy in a way that works for CPython reviewers and gives more satisfactory results for you, I would recommend that you engage more before making edits: create tickets with specific points to be improved, and/or discuss these points on discuss.  You could even join the Documentation community team to take part in tackling the big problems in Python’s doc! https://docs-community.readthedocs.io/en/latest/community/index.html

File: Doc/library/http.server.rst (hunk 4 above)
Comment (geryogam) [reply Doc/library/http.server.rst:103]: It is not the *path* component of a URI whose syntax is defined in [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3), it is the *request-target* component of an HTTP request line whose syntax is defined in [RFC 7230](https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1). The other documented attributes `command` (*method*) and `request_version` (*HTTP-version*) of `BaseHTTPRequestHandler` already use the syntactic terminology of an HTTP request line.

File: Doc/library/http.server.rst (hunk 5 above)
Comment (geryogam) [reply Doc/library/http.server.rst:109]: I am using [RFC 7230](https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1) terminology.

Comment (akulakov) [reply Doc/library/http.server.rst:109]: I also think the original version is more readable and is consistent with RFC, and provides additional information that it is a string.

**Comment by AA-Turner** [comment]
Closing due to inactivity. If you'd like to have this change merged, please update per Éric and Erlend's requests.

A

**Pull request closed without merging**